### PR TITLE
KAA-1613: Using Kaa C++ SDK, BeagleBone. [master]

### PR DESCRIPTION
**Pull request by hunch7**

### CODE REVIEW

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/SDK-Beaglebone/index.md [hunk 1]
```diff
@@ -10,103 +10,127 @@ sort_idx: 30
...
+   tar -xvzf avro-cpp-1.7.5.tar.gz -C ./avro/host
+   cd ./avro/host/avro-cpp-1.7.5
+   mkdir build && cd build
+   
+   cmake ..
+   make
+   sudo make install
+   ```
+   
+    Patch and and install Avro for target machine.
```
Comment (Polinaa) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/SDK-Beaglebone/index.md]: and and

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/SDK-Beaglebone/index.md [hunk 2]
```diff
@@ -10,103 +10,127 @@ sort_idx: 30
...
+   cd "${BB_ROOT}"
+   wget -O boost_1_59_0.tar.bz2 http://sourceforge.net/projects/boost/files/boost/1.59.0/boost_1_59_0.tar.bz2/download
+   tar -xvf boost_1_59_0.tar.bz2
+   cd boost_1_59_0
+   ./bootstrap.sh
+   sed -r 's#(using\s+gcc)(\s+;)#\1 : arm : '${CROSS}c++'\2#g' -i  project-config.jam
+   ./b2 install --prefix="${SYSROOT}"
+   ```
+
+6. Install [Apache Avro](https://avro.apache.org/) for the host and target machnies.
```
Comment (Polinaa) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/SDK-Beaglebone/index.md]: machines

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/SDK-Beaglebone/index.md [hunk 3]
```diff
@@ -10,103 +10,127 @@ sort_idx: 30
...
-        cd Botan-1.11.27
-        ./configure.py --cpu=arm --cc-bin="${CROSS}"g++ --prefix="${SYSROOT}"
-        make && make install
+After you installed the required dependencies and built the C++ SDK, you can build and run your [Kaa application]({{root_url}}Glossary/#kaa-application).
 
-## Create application
-Now, the toolchain and all dependencies are installed, and it is time to create Kaa application.
-Since BeagleBone (BeagleBone Black) is running Linux, you can refer to [the Linux guide]({{root_url}}Programming-guide/Using-Kaa-endpoint-SDKs/C++/SDK-Linux/#cpp-sdk-build) for the detailed process of application creation.
-The [CMAKE_CXX_COMPILER](https://cmake.org/cmake/help/v3.0/variable/CMAKE_LANG_COMPILER.html) specifies C++ compiler to use and [CMAKE_FIND_ROOT_PATH](https://cmake.org/cmake/help/v3.0/variable/CMAKE_FIND_ROOT_PATH.html) tells CMake where to look for required libraries which have been installed in previous step.
+Since BeagleBone (BeagleBone Black) runs on Linux, you can use the [Linux guide]({{root_url}}Programming-guide/Using-Kaa-endpoint-SDKs/C++/SDK-Linux/#build-c-sdk) to build and run your application.
```
Comment (Polinaa) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/SDK-Beaglebone/index.md]: #build-kaa-application in C++ Linux guilde

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/SDK-Beaglebone/index.md (hunk 1 above)
Comment (hunch7) [reply doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/SDK-Beaglebone/index.md]: Fixed.

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/SDK-Beaglebone/index.md (hunk 2 above)
Comment (hunch7) [reply doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/SDK-Beaglebone/index.md]: Fixed.

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/SDK-Beaglebone/index.md (hunk 3 above)
Comment (hunch7) [reply doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/SDK-Beaglebone/index.md]: Fixed.

**Review by Polinaa — APPROVED** [review]

**Review by vadimol — APPROVED** [review]

**Pull request merged**